### PR TITLE
Allow static build of rai_node

### PR DESCRIPTION
**Pull request by devinus**
~~Observes the CMake BUILD_SHARED_LIBS variable when building rai_lib.~~

~~See: https://cmake.org/cmake/help/v3.0/variable/BUILD_SHARED_LIBS.html#variable:BUILD_SHARED_LIBS~~

**Edit:** I've updated the build definition to continue to build all executables static and build our separate `rai_lib` shared library.

### CODE REVIEW

**Comment by androm3da** [comment]
But we don't want to build all of them shared, just `rai_lib`, right?

Does the build and test work as expected with both BUILD_SHARED_LIBS=ON and BUILD_SHARED_LIBS=OFF?

**Comment by devinus** [comment]
@androm3da You're absolutely right. Using `BUILD_SHARED_LIBS` for this isn't the right call.

File: rai/secure.cpp [hunk 1]
```diff
@@ -10,8 +10,6 @@
 
 #include <ed25519-donna/ed25519.h>
 
-thread_local CryptoPP::AutoSeededRandomPool rai::random_pool;
```
Comment (cryptocode) [inline rai/secure.cpp]: This is removed because it's unused?

Comment (devinus) [reply rai/secure.cpp]: @cryptocode It's in ria_lib, so it's a duplicate symbol.

**Comment by lukealonso** [comment]
Can we make this optional, since it'll double the already slow build times?

File: rai/secure.cpp (hunk 1 above)
Comment (clemahieu) [reply rai/secure.cpp]: This is here because of how windows does symbols in a shared library.  Need to make sure Windows still builds if this is removed.

**Comment by androm3da** [comment]
We're only adding this for the sake of distributing binaries?  What if we solved this problem by building distro-specific tarball releases?

**Comment by devinus** [comment]
@lukealonso It only links the already built object files twice, not rebuilds the sources twice.

**Comment by devinus** [comment]
Note: Travis failures are unrelated to this PR.

`/home/travis/build/clemahieu/raiblocks/rai/node/rpc.hpp:7:27: fatal error: boost/beast.hpp: No such file or directory`

@drizzt

**Comment by lukealonso** [comment]
@devinus could you rebase this on master? CI should work now.

**Pull request merged**